model-optimizer/extensions/middle/ONNXResize11ToInterpolateV4.py

```diff
@@ -0,0 +1,166 @@
+        begin_dim = get_height_dim(layout, input_rank)
+        end_dim = get_width_dim(layout, input_rank) + 1
+    else:
+        begin_dim = get_depth_dim(layout, input_rank)
+        end_dim = get_width_dim(layout, input_rank) + 1
+
+    sizes_ss = create_op_with_const_inputs(graph, StridedSlice,
+                                           {1: int64_array([begin_dim]),
+                                            2: int64_array([end_dim]),
+                                            3: int64_array([1])},
+                                           {'name': resize_name + '/StridedSlice_',
+                                            'begin_mask': int64_array([1]),
+                                            'end_mask': int64_array([1]),
+                                            'new_axis_mask': int64_array([0]),
+                                            'shrink_axis_mask': int64_array([0]),
+                                            'ellipsis_mask': int64_array([0])})
+    scales_ss = create_op_with_const_inputs(graph, StridedSlice,
+                                            {1: int64_array([begin_dim]),
+                                             2: int64_array([end_dim]),
+                                             3: int64_array([1])},
+                                            {'name': resize_name + '/StridedSlice_',
+                                             'begin_mask': int64_array([1]),
+                                             'end_mask': int64_array([1]),
+                                             'new_axis_mask': int64_array([0]),
+                                             'shrink_axis_mask': int64_array([0]),
+                                             'ellipsis_mask': int64_array([0])})
+    axes_node = Const(graph,
+                      {'name': resize_name + '/axis_',
+                       'value': int64_array(np.arange(begin_dim, end_dim))}).create_node()
+
+    shape_calculation_mode = 'scales' if num_of_inputs == 3 else 'sizes'
+
+    interpolate_node = Interpolate(graph, {'version': 'opset4',
+                                           'mode': convert_mode(resize.mode),
+                                           'coordinate_transformation_mode': resize.coordinate_transformation_mode,
+                                           'cube_coeff': resize.cube_coeff,
+                                           'nearest_mode': resize.nearest_mode,
+                                           'pads_begin': int64_array([0]),
+                                           'pads_end': int64_array([0]),
+                                           'antialias': 0,
+                                           'shape_calculation_mode': shape_calculation_mode,
+                                           'in_ports_count': 4}).create_node()
+
+    axes_node.out_port(0).connect(interpolate_node.in_port(3))
+    shape_of = Shape(graph, {'name': resize_name + '/ShapeOf_'}).create_node()
+
+    if num_of_inputs == 3:
```

Comment:
Do you have an example of ONNX model where the Resize has set `sizes` input and not `scales`? What is the number of inputs in this case? I am concerned about this case: "If 'size' is needed, the user can use an empty string as the name of 'scales' in this operator's input list.". How the model will look in this case?

Reply:
According to ONNX Resize-11 specification, https://github.com/onnx/onnx/blob/rel-1.7.0/docs/Operators.md#Resize, `sizes` input is optional. Hence, we should use `scales`, if there are 3 inputs.